modules/reference/services/proxy-server/3proxy-config.nix

```diff
@@ -0,0 +1,70 @@
+# Copyright 2024 TII (SSRC) and the Ghaf contributors
+# SPDX-License-Identifier: Apache-2.0
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  cfg = config.ghaf.reference.services.proxy-server;
+  inherit (lib) mkEnableOption mkIf;
+  # use nix-prefetch-url to calculate sha256 checksum
+  msEndpointsFile = pkgs.fetchurl {
+    url = "https://endpoints.office.com/endpoints/worldwide?clientrequestid=b10c5ed1-bad1-445f-b386-b919946339a7";
```

Comment:
Thanks for the comment! We will have additional PR to replace 3proxy with mitmproxy, which will be fixed there. I will add your comment as TODO.

Reply:
The URL contents have been added as a file to the source tree, and it will not disrupt the Ghaf builds. The URLs do not change very frequently. If we used IP addresses, it could have been the problem.

Reply:
URL updates are done at the end of each month. The last update was 30 Sep and next one expected 31 Oct (or close to it)